src/web-auth/index.js

```diff
@@ -303,106 +307,139 @@ WebAuth.prototype.validateAuthenticationResponse = function(
       errorDescription: '`state` does not match.'
     });
   }
+
   var transactionNonce =
     options.nonce || (transaction && transaction.nonce) || null;
 
+  var transactionOrganization = transaction && transaction.organization;
   var appState = options.state || (transaction && transaction.appState) || null;
 
   var callback = function(err, payload) {
     if (err) {
       return cb(err);
     }
+
     if (transaction && transaction.lastUsedConnection) {
       var sub;
+
       if (payload) {
         sub = payload.sub;
       }
+
       _this.ssodataStorage.set(transaction.lastUsedConnection, sub);
     }
+
     return cb(null, buildParseHashResponse(parsedHash, appState, payload));
   };
 
   if (!parsedHash.id_token) {
     return callback(null, null);
   }
-  return this.validateToken(parsedHash.id_token, transactionNonce, function(
-    validationError,
-    payload
-  ) {
-    if (!validationError) {
-      if (!parsedHash.access_token) {
-        return callback(null, payload);
-      }
-      // id_token's generated by non-oidc applications don't have at_hash
-      if (!payload.at_hash) {
-        return callback(null, payload);
-      }
-      // here we're absolutely sure that the id_token's alg is RS256
-      // and that the id_token is valid, so we can check the access_token
-      return new IdTokenVerifier().validateAccessToken(
-        parsedHash.access_token,
-        'RS256',
-        payload.at_hash,
-        function(err) {
-          if (err) {
-            return callback(error.invalidToken(err.message));
+
+  return this.validateToken(
+    parsedHash.id_token,
+    transactionNonce,
+    function(validationError, payload) {
+      if (!validationError) {
+        // Verify the organization
+        if (transactionOrganization) {
+          if (!payload.org_id) {
+            return callback(
+              error.invalidToken(
+                'Organization Id (org_id) claim must be a string present in the ID token'
+              )
+            );
+          }
+
+          if (payload.org_id !== transactionOrganization) {
+            return callback(
+              error.invalidToken(
+                'Organization Id (org_id) claim value mismatch in the ID token; expected "' +
+                transactionOrganization +
+                  '", found "' +
+                  payload.org_id +
+                  '"'
+              )
+            );
           }
+        }
```

Comment:
This could have gone into `validateToken`, but putting it here has the advantage that we don't then need to make a breaking change to that method interface, which might be being used by a consumer of the SDK.